test/common/stats/stat_test_utility.h

```diff
@@ -72,6 +77,79 @@ class MemoryTest {
   const size_t memory_at_construction_;
 };
 
+// Helper class to use in lieu of an actual Stats::Store for doing lookups by
+// name. The intent is to remove the deprecated Scope::counter(const
+// std::string&) methods, and always use this class for accessing stats by
+// name.
+//
+// This string-based lookup wrapper is needed because the underlying name
+// representation, StatName, has multiple ways to represent the same string,
+// depending on which name segments are symbolic (known at compile time), and
+// which are dynamic (e.g. based on the request, e.g. request-headers, ssl
+// cipher, grpc method, etc). We need to do name-matching based only on strings
+// to avoid having the tests know which segments are dynamic.
+//
+// Note that this class does not implement the of Stats::Store interface, but
+// provides similar methods used for lookup by name, as well as an accessor
+// to the underlying store.
+class TestStore {
+public:
+  // Constructs a TestStore with an owned IsolatedStoreImpl. This is useful
+  // for replacing the IsolatedStoreImpl in a unit-test with this class, with
+  // minimal code changes, other than accessing the underlying store with
+  // ".store()".
+  TestStore() : owned_storage_(std::make_unique<IsolatedStoreImpl>()), store_(*owned_storage_) {}
+
+  // Constructs a store using a symbol table, allowing for explicit sharing.
+  explicit TestStore(SymbolTable& symbol_table)
+      : owned_storage_(std::make_unique<IsolatedStoreImpl>(symbol_table)), store_(*owned_storage_) {
+  }
+
+  // Constructs a TestStore with an already-existing
+  // IsolatedStatStoreImpl. This is useful when a test is built using a mock
+  // store, or we are referencing the local store held in an Upstream.
+  explicit TestStore(Store& store) : store_(store) {}
+
+  OptionalCounter findCounter(const std::string& name) {
+    CounterSharedPtr counter = TestUtility::findCounter(store_, name);
+    return counter == nullptr ? OptionalCounter() : *counter;
+  }
+  Counter& counter(const std::string& name) {
+    OptionalCounter opt = findCounter(name);
+    RELEASE_ASSERT(opt, absl::StrCat("could not find counter ", name));
+    return const_cast<Counter&>(opt->get());
+  }
+  OptionalGauge findGauge(const std::string& name) {
+    GaugeSharedPtr gauge = TestUtility::findGauge(store_, name);
+    return gauge == nullptr ? OptionalGauge() : *gauge;
+  }
+  Gauge& gauge(const std::string& name) {
+    OptionalGauge opt = findGauge(name);
+    RELEASE_ASSERT(opt, absl::StrCat("could not find gauge ", name));
+    return const_cast<Gauge&>(opt->get());
+  }
+
+  // TODO(jmarantz): support for histograms will be added when needed as we
```

Comment:
I would just delete this comment, or perhaps just the commented out code?

Reply:
actually I just made it work & tested it.